bundles/org.openhab.core.model.rule.runtime/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleEngineImpl.java

```diff
@@ -83,11 +86,10 @@
     private final ModelRepository modelRepository;
     private final ScriptEngine scriptEngine;
 
-    private RuleTriggerManager triggerManager;
+    private @NonNullByDefault({}) Injector injector;
+    private @NonNullByDefault({}) RuleTriggerManager triggerManager;
```

Comment:
Can't we move the two lines https://github.com/openhab/openhab-core/blob/ca1cda3f7521d56b85c4ee87a7417ab5aeb4e6f4/bundles/org.openhab.core.model.rule.runtime/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleEngineImpl.java#L109-L110 to the constructor and drop the additional `@NonNullByDefault({})`.
There is also no need set `triggerManager` to `null` on deactivation (it is already cleared).

Or do you prefer to separate the changes, too?

Reply:
Yes, that should be possible. I would like to do that in a separate PR because there are quite some `null` checks in the code on the `triggerManager` which can be removed in advance.